diff --git a/compile.c b/compile.c
index 0ddb290..9fd8cb3 100644
--- a/compile.c
+++ b/compile.c
@@ -12,7 +12,7 @@
 #include "ruby/internal/config.h"
 #include <math.h>
 
-#ifdef HAVE_DLADDR
+#if defined(HAVE_DLADDR) && !defined(__MVS__)
 # include <dlfcn.h>
 #endif
 
@@ -11506,7 +11506,7 @@ insn_data_to_s_detail(INSN *iobj)
               case TS_FUNCPTR:
                 {
                     void *func = (void *)OPERAND_AT(iobj, j);
-#ifdef HAVE_DLADDR
+#if defined(HAVE_DLADDR) && !defined(__MVS__)
                     Dl_info info;
                     if (dladdr(func, &info) && info.dli_sname) {
                         rb_str_cat2(str, info.dli_sname);
