diff --git a/cont.c b/cont.c
index eb9a9b4..b147782 100644
--- a/cont.c
+++ b/cont.c
@@ -41,6 +41,7 @@ extern int madvise(caddr_t, size_t, int);
 #include "vm_sync.h"
 #include "id_table.h"
 #include "ractor_core.h"
+#include <zos-base.h>
 
 static const int DEBUG = 0;
 
@@ -481,8 +482,11 @@ fiber_pool_allocate_memory(size_t * count, size_t stride)
 #else
         errno = 0;
         size_t mmap_size = (*count)*stride;
+#ifdef __MVS__
+        void * base = anon_mmap(NULL, mmap_size);
+#else
         void * base = mmap(NULL, mmap_size, PROT_READ | PROT_WRITE, FIBER_STACK_FLAGS, -1, 0);
-
+#endif
         if (base == MAP_FAILED) {
             // If the allocation fails, count = count / 2, and try again.
             *count = (*count) >> 1;
