diff --git a/process.c b/process.c
index 5fbc9be..90c24c4 100644
--- a/process.c
+++ b/process.c
@@ -8594,7 +8594,7 @@ rb_clock_getres(int argc, VALUE *argv, VALUE _)
     timetick_int_t denominators[2];
     int num_numerators = 0;
     int num_denominators = 0;
-#ifdef HAVE_CLOCK_GETRES
+#if defined(HAVE_CLOCK_GETRES) || defined(__MVS__)
     clockid_t c;
 #endif
 
@@ -8697,11 +8697,12 @@ rb_clock_getres(int argc, VALUE *argv, VALUE _)
 #endif
     }
     else if (NUMERIC_CLOCKID) {
-#if defined(HAVE_CLOCK_GETRES)
+#if defined(HAVE_CLOCK_GETRES) || defined(__MVS__)
         struct timespec ts;
         c = NUM2CLOCKID(clk_id);
       getres:
-        ret = clock_getres(c, &ts);
+        // ret = clock_getres(c, &ts);
+        ret = -1;
         if (ret == -1)
             clock_failed("getres", errno, clk_id);
         tt.count = (int32_t)ts.tv_nsec;
