diff --git a/ruby_assert.h b/ruby_assert.h
index d8f7ddd..9f000cb 100644
--- a/ruby_assert.h
+++ b/ruby_assert.h
@@ -7,8 +7,13 @@
  *             modify this file, provided that  the conditions mentioned in the
  *             file COPYING are met.  Consult the file for details.
  */
-#include "ruby/assert.h"
+#ifdef assert
+# undef assert
+#endif
+#define assert assert_undefine
+#include <assert.h>
 #undef assert
+#include "ruby/assert.h"
 #define assert RUBY_ASSERT_NDEBUG
 
 #endif /* RUBY_TOPLEVEL_ASSERT_H */
