diff --git a/thread_pthread.c b/thread_pthread.c
index 423814c..42992d5 100644
--- a/thread_pthread.c
+++ b/thread_pthread.c
@@ -2022,7 +2022,7 @@ native_thread_init_stack(rb_thread_t *th, void *local_in_parent_frame)
     th->ec->machine.asan_fake_stack_handle = asan_get_thread_fake_stack_handle();
 #endif
 
-    if (!native_main_thread.id) {
+    if (!native_main_thread.id.__) {
         /* This thread is the first thread, must be the main thread -
          * configure the native_main_thread object */
         native_thread_init_main_thread_stack(local_in_parent_frame);
